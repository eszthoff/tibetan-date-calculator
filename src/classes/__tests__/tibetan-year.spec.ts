import MockDate from 'mockdate';
import TibetanYear from '../tibetan-year';

describe('TibetanYear', () => {
  beforeAll(() => {
    MockDate.set('2019/06/07');
  });
  afterAll(() => {
    MockDate.reset();
  });

  it('should create a correct class when invoked without arg', () => {
    expect(new TibetanYear()).toMatchSnapshot();
  });
  it('should create a correct class when invoked with number arg', () => {
    expect(new TibetanYear(2145)).toMatchSnapshot();
  });
  it('should create a correct class when invoked with number arg and western set to true', () => {
    expect(new TibetanYear(2013, true)).toMatchSnapshot();
  });
  it('should create a correct class when invoked with object arg', () => {
    expect(new TibetanYear({
      rabjungCycle: 17, rabjungYear: 33
    })).toMatchSnapshot();
  });
  it('should not calculate months when initially incoved', () => {
    const year = new TibetanYear();
    expect(year.months.length).toEqual(0);
  });
  it('should not calculate months when requested', () => {
    const year = new TibetanYear();
    const months = year.getMonths();

    expect(year.months.length).not.toEqual(0);
    expect(months.length).toEqual(13);
  });
  it('should return the correct string when `toRabjungString` is called', () => {
    const year = new TibetanYear();
    expect(year.toRabjungString()).toEqual('The 33. year of the 17. rabjung cycle');
  });
  it('should return the correct string when `toString` is called', () => {
    const year = new TibetanYear();
    expect(year.toString()).toEqual('2146');
  });
});
