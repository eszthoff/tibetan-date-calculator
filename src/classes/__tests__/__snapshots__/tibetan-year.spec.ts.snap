// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TibetanYear should create a correct class when invoked with number arg 1`] = `
TibetanYear {
  "animal": "Dog",
  "element": "Earth",
  "gender": "Male",
  "months": Array [],
  "rabjungCycle": 17,
  "rabjungYear": 32,
  "tibYearNum": 2145,
  "westernYear": 2018,
}
`;

exports[`TibetanYear should create a correct class when invoked with number arg and western set to true 1`] = `
TibetanYear {
  "animal": "Snake",
  "element": "Water",
  "gender": "Female",
  "months": Array [],
  "rabjungCycle": 17,
  "rabjungYear": 27,
  "tibYearNum": 2140,
  "westernYear": 2013,
}
`;

exports[`TibetanYear should create a correct class when invoked with object arg 1`] = `
TibetanYear {
  "animal": "Pig",
  "element": "Earth",
  "gender": "Female",
  "months": Array [],
  "rabjungCycle": 17,
  "rabjungYear": 33,
  "tibYearNum": 2146,
  "westernYear": 2019,
}
`;

exports[`TibetanYear should create a correct class when invoked without arg 1`] = `
TibetanYear {
  "animal": "Pig",
  "element": "Earth",
  "gender": "Female",
  "months": Array [],
  "rabjungCycle": 17,
  "rabjungYear": 33,
  "tibYearNum": 2146,
  "westernYear": 2019,
}
`;
